client/my-sites/site-settings/form-general.jsx

```diff
@@ -577,6 +581,55 @@ export class SiteSettingsFormGeneral extends Component {
 		);
 	}
 
+	isLaunchable() {
+		const { isSiteOnECommerceTrial, isSiteOnMigrationTrial } = this.props;
+		return ! isSiteOnECommerceTrial && ! isSiteOnMigrationTrial;
```

Comment:
I'm imagining in the future when we have more trials, replicating this logic in a lot of places may be clumsy. I was wondering if we can have a global selector called `isTrialSite` somewhere.

Am ok with the current implementation, but could be nice to extract it somewhere.

Reply:
@vishnugopal I've added it [here](https://github.com/Automattic/wp-calypso/pull/79921/files#diff-60deb7cd9d038ecab322bcb57488d1ff0b8f1b14e15f4cbeab1d1bcdd399a2b7R11-R13). 🙂

Reply:
> I'm imagining in the future when we have more trials, replicating this logic in a lot of places may be clumsy. I was wondering if we can have a global selector called isTrialSite somewhere.

Thanks for the suggestion @vishnugopal. This makes perfect sense! I did some refactor then I saw @zaerl also haas this `isTrialPlan ` selector just now but looks like there's still some discussions going on. I can hold this PR for now util everything is clear.

Reply:
I can simply remove it and use yours, not a problem. My diff is small enough to just use your new function in the only line when I do this check. Feel free to land this.

Reply:
> I can simply remove it and use yours; not a problem. My diff is small enough to just use your new function in the only line where I do this check. Feel free to land this.

That works too, thanks @zaerl ! Could you give it a review when you have some time since I've refactored it again and I'd like to make sure it's still working like it is now. Thanks!

Reply:
> That works too, thanks @zaerl ! Could you give it a review when you have some time since I've refactored it again and I'd like to make sure it's still working like it is now. Thanks!

The PR is good! I see some failed unit tests, but it's just timeouts, so no problem for us.

Reply:
The new plan grid is about to be moved to a new package. So they [asked me](https://github.com/Automattic/wp-calypso/pull/79921#discussion_r1277174874) if the `isTrialSite` function can be the first of the list of functions to be moved in another place.

@vishnugopal I put [my PR](https://github.com/Automattic/wp-calypso/pull/79921) on hold because it's not a blocker. IMHO, there is a better time to start a refactoring of these selectors. The `Plan` is moved through various components, and we need to identify when/where the trial functions are called.

This could be a blocker for our project, and we could concentrate on our uphill tasks.